internal/cmd/manager/controller/controller.go

```diff
@@ -314,6 +307,20 @@ func RunController(
 	return nil
 }
 
+func getNamespacesToWatch(conf *configuration.Data) map[string]cache.Config {
+	if len(conf.WatchNamespace) == 0 {
+		setupLog.Info("Listening for changes on all namespaces")
+		return nil
+	}
+	namespaces := map[string]cache.Config{}
+	for _, namespace := range conf.WatchedNamespaces() {
+		namespaces[namespace] = cache.Config{}
+	}
```

Comment:
This won't watch any more the namespace in which the operator is installed unless explicitly asked by the user, in case of a multi-namespace setup in openshift. Cluster creation will fail because the operator will try to copy secrets from its own namespace to the cluster's.


We could use something like
```suggestion
	for _, namespace := range conf.WatchedNamespaces() {
		namespaces[namespace] = cache.Config{}
	}
	namespaces[conf.OperatorNamespace] = cache.Config{}
```
implicitly adding the operator namespace to the watched ones.

Reply:
It wont because the function will exit if 
```
	if len(conf.WatchNamespace) == 0 {
		setupLog.Info("Listening for changes on all namespaces")
		return nil
	}
```